docs/pages/guides/prebuild.md

```diff
@@ -0,0 +1,168 @@
+
+Due to the pivotal nature of the flag, you'll be warned to have a clean git status when the `--clean` flag is used. This prompt is optional and will be skipped when encountered in CI.
+
+If you'd like, you can disable the check by enabling the environment variable `EXPO_NO_GIT_STATUS=1`.
+
+The purpose of the prompt is to encourage managed workflow users to add the `/ios` and `/android` folders to the project's `.gitignore`, ensuring that the project is always managed. However, this can make custom config plugins harder to develop so we haven't introduced any mechanism to enforce this behavior.
+
+Advanced projects may need to swap between workflows often, building custom functionality natively in Xcode and Android Studio, then moving that functionality into local Config Plugins, and repeating. As the Expo Config Plugin ecosystem matures, the need to develop plugins like this will be drastically reduced.
+
+<!-- It is also theoretically possible to make clean builds take seconds rather than minutes, meaning `--clean` could become the default behavior in the future. -->
+
+## Templates
+
+You can customize how the native folders are generated while remaining in the managed workflow by building [Expo Config Plugins][config-plugins]. Many config plugins already exist for lots of modifications. You can see the [list][config-plugins-repo] for more information.
+
+Prebuild generates template files before modifying them with Expo Config Plugins. The template files are versioned and come from the npm package [`expo-template-bare-minimum`][template]. You can change the template used by passing `--template /path/to/template.tgz` to the `npx expo prebuild` command. This is not recommended as the base modifiers in `@expo/prebuild-config` make some undocumented assumptions about the template files.
+
+## Side-Effects
+
+As of SDK 46, `npx expo prebuild` has some unfortunate side-effects that we plan to remove. Ideally, running `npx expo prebuild` would only generate an `ios` and `android` folder but this is currently not the case.
+
+In addition to generating the native folders, prebuild also makes the following modifications:
+
+- Generating a `metro.config.js` file.
+- Generating an `index.js` file.
+- Removing the `main` field in the `package.json`.
+- Modifying the `scripts` field in the `package.json`.
+- Modifying the `dependencies` field in the `package.json`.
+
+The convenience change to the `scripts` field is the only thing that alters how a user develops their app before/after prebuild, all other changes can be left as-is if desired.
+
+## Optionality
+
+Prebuilding is completely optional and does not disqualify a project from using other features or services offered by Expo. We created the designation _bare workflow_ to refer projects that do not use `npx expo prebuild`.
```

Comment:
> disposing of them when no longer needed as with prebuild.

This part isn't relevant, we want to avoid the distinction that the existence of files changes anything. All that matters is if the user continues to run prebuild or not.

Reply:
```suggestion
Prebuilding is completely optional and it works great with all tools and services offered by Expo. We created the designation _bare workflow_ to refer projects that do not use `npx expo prebuild` — projects where developers make direct changes to their native projects, rather than continuously generating them on demand as with prebuild.
```